File: packages/agent-tui/_examples/stdio-typescript/agent.ts
#!/usr/bin/env npx tsx
/** Minimal agent-tui stdio backend. Zero dependencies beyond Node. */
import * as readline from 'readline';

const rl = readline.createInterface({ input: process.stdin });

function write(obj: any) {
  process.stdout.write(JSON.stringify(obj) + '\n');
}

function notify(params: any) {
  write({ jsonrpc: '2.0', method: 'stream.event', params });
}

rl.on('line', (line) => {
  const req = JSON.parse(line);
  const { method, params = {}, id } = req;

  if (method === 'listAgents') {
    write({ jsonrpc: '2.0', result: [{ id: 'default', name: 'Assistant', role: 'Helpful' }], id });
  } else if (method === 'createConversation') {
    write({ jsonrpc: '2.0', result: { id: 'conv-1', agent_id: params.agent_id ?? 'default' }, id });
  } else if (method === 'sendMessage') {
    for (const word of `You said: ${params.content}`.split(' ')) {
      notify({ type: 'message.delta', data: { delta: word + ' ' } });
    }
    notify({ type: 'done', data: {} });
    write({ jsonrpc: '2.0', result: { status: 'complete' }, id });
  } else {
    write({ jsonrpc: '2.0', error: { code: -32601, message: `Unknown: ${method}` }, id });
  }
});

File: packages/agentic-tui/_examples/stdio-python/agent.py
#!/usr/bin/env python3
"""Minimal agentic-tui stdio backend. Zero dependencies."""
import sys, json

def handle(method, params, req_id):
    if method == "listAgents":
        return {"result": [{"id": "default", "name": "Assistant", "role": "Helpful assistant"}], "id": req_id}
    elif method == "createConversation":
        return {"result": {"id": "conv-1", "agent_id": params.get("agent_id", "default")}, "id": req_id}
    elif method == "sendMessage":
        content = params["content"]
        for word in f"You said: {content}".split():
            notify({"type": "message.delta", "data": {"delta": word + " "}})
        notify({"type": "done", "data": {}})
        return {"result": {"status": "complete"}, "id": req_id}
    else:
        return {"error": {"code": -32601, "message": f"Method not found: {method}"}, "id": req_id}

def notify(params):
    write({"jsonrpc": "2.0", "method": "stream.event", "params": params})

def write(obj):
    print(json.dumps(obj), flush=True)

for line in sys.stdin:
    req = json.loads(line.strip())
    resp = handle(req["method"], req.get("params", {}), req.get("id"))
    if resp:
        write({"jsonrpc": "2.0", **resp})
